documentation/src/docs/wire-protocol.md

```diff
@@ -11,102 +11,97 @@ You may obtain a copy of the License at
+1.  `ConnectionId`: UUID (16 bytes).
+2.  `ConnectionOffsetAckLevel`: Long (8 bytes). The highest offset before which all the data is successfully stored on this Stream Segment for this `connectionId`.
 
-When appending a client
+When appending a client:
 
-1.  Establishes a connection to what it thinks is the correct host.
-2.  Sends a Setup Append request.
-3.  Waits for the Append Setup reply.
+- Establishes a connection to host chosen by it.
+- Sends a `Setup Append` request.
+- Waits for the `Append Setup reply`.
 
-Then it can
-1.  Send a BeginEventBlock request
-2.  Send as many messages as can fit in the block
-3.  Send an EndEventBlock request
+After receiving the `Append Setup` reply, it performs the following:
+- Send a `BeginEventBlock` request.
+- Send as many messages that can fit in the block.
+- Send an `EndEventBlock` request.
 
-While this is happening the server will be periodically sending it DataAppended replies acking messages. Note that there can be multiple "Appends" setup for a given TCP connection. This allows a client to share a connection when producing to multiple segments.
+While this is happening, the server will be periodically sending it `DataAppended` replies acking messages. Note that there can be multiple `Appends Setup` for a given TCP connection. This allows a client to share a connection when producing to multiple Segments.
 
-A client can optimize its appending by specifying a large value in it's BeginAppendBlock message, as the events inside of the block do not need to be processed individually.
+A client can optimize its appending by specifying a large value in it's `BeginAppendBlock` message, as the events inside of the block do not need to be processed individually.
 
-The EndEventBlock message specifies the size of the append block rather than the BeginAppendBlock message. This means that the size of the data in the block need not be known in advance. This is useful if a client is producing a stream of small messages. It can begin a block, write many messages and then when it comes time to end the block, it can write a partial message followed the EndAppendBlock message, followed by the remaining partial message. This would avoid having headers on all of the messages in the block without having to buffer them in ram in its process.
+The `EndEventBlock` message specifies the size of the append block rather than the `BeginAppendBlock` message. The size of the data in the block need not be known in advance. This is useful if a client is producing a Stream of small messages. It can begin a block, write many messages and then when it comes time to end the block, it can write a partial message followed the `EndAppendBlock` message, followed by the remaining partial message. This would avoid having headers on all of the messages in the block without having to buffer them in ram in its process.
```

Comment:
@tkaitchuck @howdy
Could you rephrase this para??

Reply:
The `ApppendBlockEnd` has a `sizeOfWholeEvents` to allow the append block to be less than full. This allows the client to begin writing a block before it has a large number of events. This avoids the need to buffer up events in the client and allows for lower latency.